### PR TITLE
[MAINTENANCE] Deprecate `great_expectations.render.types` package

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- Move implementations to `great_expectations.render.components`
- Add deprecation message and allow imports from `types` to continue for now

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/render/__init__.py [hunk 1]
```diff
@@ -125,6 +126,499 @@ class LegacyDescriptiveRendererType(str, Enum):
     )
 
 
+class RenderedContent:
```
Comment (Kilo59) [inline great_expectations/render/__init__.py]: Can we move this and other class definitions out of `__init__.py`?
Maybe put them in a `render/core.py` or `render/rendered_content.py`, and import them into the `__init__.py` namespace to preserve existing imports

**Review by cdkini — APPROVED** [review]

File: great_expectations/render/__init__.py [hunk 2]
```diff
@@ -1,277 +1,34 @@
...
-from copy import deepcopy
-from enum import Enum
-from typing import List, Optional, Union
-
-from marshmallow import INCLUDE, Schema, fields, post_dump, post_load
-
-from great_expectations.types import DictDot
-
-from .types import RenderedContent
+from .components import (
```
Comment (Kilo59) [inline great_expectations/render/__init__.py:1]: 👍

File: great_expectations/render/types/__init__.py [hunk 3]
```diff
@@ -1,497 +1,246 @@
...
+    )
+
+
+class RenderedAtomicContent(RenderedAtomicContentRender):
+    # deprecated-v0.15.32
+    classname = "RenderedAtomicContent"
+    warnings.warn(
+        _get_deprecation_warning_message(classname=classname),
+        DeprecationWarning,
+    )
```
Comment (Kilo59) [inline great_expectations/render/types/__init__.py]: So will all these warnings run when this `types` package is imported?

Comment (NathanFarmer) [reply great_expectations/render/types/__init__.py]: No they would not have. Updated to add a test and leverage `__init__`.

**Pull request merged**